src/components/tx/security/shared/SecurityWarnings/index.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Alert,
+  type AlertColor,
+  SvgIcon,
+  Typography,
+  Box,
+  Grid,
+  Checkbox,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  FormControlLabel,
+} from '@mui/material'
+
+import { SecuritySeverity } from '@/services/security/modules/types'
+import AlertIcon from '@/public/images/notifications/alert.svg'
+
+import css from './styles.module.css'
+import { LoadingLabel } from '../LoadingLabel'
+import { type ReactElement } from 'react'
+import RedefineLogo from '@/public/images/transactions/redefine.svg'
+import RedefineLogoDark from '@/public/images/transactions/redefine-dark-mode.svg'
+
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
+import Track from '@/components/common/Track'
+import { MODALS_EVENTS } from '@/services/analytics'
+import { useDarkMode } from '@/hooks/useDarkMode'
+
+type SecurityWarningProps = {
+  color: AlertColor
+  label: string
+  action?: string
+}
+
+const ACTION_REJECT = 'Reject this transaction'
+const ACTION_REVIEW = 'Review before processing'
+
+export const mapSeverityComponentProps: Record<SecuritySeverity, SecurityWarningProps> = {
+  [SecuritySeverity.CRITICAL]: {
+    action: ACTION_REJECT,
+    color: 'error',
+    label: 'Critical issue',
+  },
+  [SecuritySeverity.HIGH]: {
+    action: ACTION_REJECT,
+    color: 'error',
+    label: 'High issue',
+  },
+  [SecuritySeverity.MEDIUM]: {
+    action: ACTION_REVIEW,
+    color: 'warning',
+    label: 'Medium issue',
+  },
+  [SecuritySeverity.LOW]: {
+    action: ACTION_REVIEW,
+    color: 'warning',
+    label: 'Low issue',
+  },
+  [SecuritySeverity.NONE]: {
+    color: 'info',
+    label: 'No issues found',
+  },
+}
+
+export const SecurityHint = ({ severity, warnings }: { severity: SecuritySeverity; warnings: string[] }) => {
+  const severityProps = mapSeverityComponentProps[severity]
+  const pluralizedLabel = `${severityProps.label}${warnings.length > 1 ? 's' : ''}`
+  return (
+    <>
+      <Alert
+        className={css.hint}
+        severity={severityProps.color}
+        sx={{ bgcolor: ({ palette }) => palette[severityProps.color].background }}
+        icon={
+          <SvgIcon
+            component={AlertIcon}
+            inheritViewBox
+            color={severityProps.color}
+            sx={{
+              '& path': {
+                fill: ({ palette }) => palette[severityProps.color].main,
+              },
+            }}
+          />
+        }
+      >
+        {severity !== SecuritySeverity.NONE && <Typography variant="h5">{pluralizedLabel}</Typography>}
+        <Box display="flex" flexDirection="column" gap={2}>
+          {warnings.map((warning) => (
+            <Typography key={warning} variant="body2">
+              {warning}
+            </Typography>
+          ))}
```

Comment:
If there is more than one warning, shall we add them to a list? Currently there is no clear definition between warnings. What do you think? cc @liliiaorlenko

Reply:
Good point. Here is an example of how 2 warnings look like. Semantically we should make them list elements but I would also agree to make it more visible as separate elements.
<img width="532" alt="Screenshot 2023-06-21 at 12 20 34" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/e7551178-57a2-4404-a660-e64d40a90406">

Reply:
Something like this @liliiaorlenko ?
<img width="546" alt="Screenshot 2023-06-21 at 12 36 30" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/2c63cb29-c1b8-4ce7-8630-2dcabd29db62">

Reply:
I just noticed that it's always pluralised. It should be: "1 more issue" in the "Low issue" box, for example.

Reply:
Fixed it